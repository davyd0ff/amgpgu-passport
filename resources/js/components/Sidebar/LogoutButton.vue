<template>
  <a
    href="#"
    class="black-text exit-to-app"
    v-on:click.prevent="onLogout"
    title="LOGIN_LOGOUT"
  >
    <i class="material-icons">exit_to_app</i>
  </a>
</template>

<script>
export default {
  name: 'LogoutButton',
  methods: {
    onLogout: function () {
      this.$store.dispatch('logout').then(() => {
        // route to login
      });
    },
  },
};
</script>

<style scoped></style>
