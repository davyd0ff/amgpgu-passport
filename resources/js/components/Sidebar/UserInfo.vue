<template>
  <div class="user-view">
    <div class="background">
      <img
        src="https://www.las.ru/media/foto/kabinet-rukovoditelya-seventies-foto-01-a.jpg"
      />
    </div>
    <router-link to="/profile">
      <img
        class="circle"
        src="https://sun9-11.userapi.com/impf/c638426/v638426062/50419/eNL3yaz6niM.jpg?size=300x300&quality=96&proxy=1&sign=281eca8b1027b3a552a1e591fd6d5019&type=audio"
      />
    </router-link>
    <router-link to="/profile">
      <span class="black-text name">{{ username }}</span>
    </router-link>
    <router-link to="/profile">
      <span class="black-text email">{{ email }}</span>
    </router-link>
    <logout-button />
  </div>
</template>

<script>
import LogoutButton from './LogoutButton';
export default {
  name: 'UserInfo',
  components: { LogoutButton },
  computed: {
    info: function () {
      return this.$store.getters.user;
    },
    username: function () {
      if (this.info && this.info.lastname) {
        return `${this.info.lastname} ${this.info.firstname} ${this.info.middlename}`;
      }
      return '';
    },
    email: function () {
      if (this.info && this.info.email) {
        return this.info.email;
      }
      return '';
    },
  },
};
</script>

<style scoped></style>
