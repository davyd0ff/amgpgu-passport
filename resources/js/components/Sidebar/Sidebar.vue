<template>
  <aside>
    <ul
      id="sidenav-left"
      class="sidenav sidenav-fixed grey-text text-darken-4"
      ref="sidenav"
    >
      <li>
        <user-info />
      </li>
      <li class="no-padding">
        <ul ref="collapsible" class="collapsible collapsible-accordion">
          <student-menu v-if="isStudent" v-bind:menu="menu.student" />
          <listener-menu v-if="isListener" v-bind:menu="menu.listener" />
          <admin-menu v-if="hasAccessAdminMenu" v-bind:menu="menu.admin" />
        </ul>
      </li>
    </ul>
  </aside>
</template>

<script>
import M from 'materialize-css';

import StudentMenu from '@/components/Menu/StudentMenu';
import ListenerMenu from '@/components/Menu/ListenerMenu';
import AdminMenu from '@/components/Menu/AdminMenu';
import UserInfo from './UserInfo';

export default {
  name: 'Sidebar',
  components: { AdminMenu, ListenerMenu, StudentMenu, UserInfo },

  computed: {
    menu: function () {
      return this.$store.getters.menu;
    },
    isStudent: function () {
      return this.menu?.student?.length > 0 || false;
    },
    isListener: function () {
      return this.menu?.listener?.length > 0 || false;
    },
    hasAccessAdminMenu: function () {
      return this.menu?.admin?.length > 0 || false;
    },
  },

  mounted() {
    this.$sidenav = M.Sidenav.init(this.$refs.sidenav);
    this.$collapsible = M.Collapsible.init(this.$refs.collapsible);
    // M.AutoInit();
  },
  destroyed() {
    this.$sidenav.destroy();
    this.$collapsible.destroy();
  },
};
</script>

<style scoped></style>
