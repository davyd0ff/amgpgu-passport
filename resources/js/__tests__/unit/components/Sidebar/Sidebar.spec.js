import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import Sidebar from '@/components/Sidebar/Sidebar.vue';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Sidebar.vue', () => {
  it('render', () => {
    const store = new Vuex.Store();

    const wrapper = shallowMount(Sidebar, {
      store,
      localVue,
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  it('computed property isStudent is truthy, others are Falsy', () => {
    const store = new Vuex.Store({
      state: {
        menu: {
          student: ['TEST'],
        },
      },
      getters: {
        menu: (state) => {
          return state.menu;
        },
      },
    });

    const wrapper = shallowMount(Sidebar, {
      store,
      localVue,
    });

    expect(wrapper.vm.isListener).toBeFalsy();
    expect(wrapper.vm.hasAccessAdminMenu).toBeFalsy();
    expect(wrapper.vm.isStudent).toBeTruthy();
  });

  it('computed property isListener is truthy, others are Falsy', () => {
    const store = new Vuex.Store({
      state: {
        menu: {
          listener: ['TEST'],
        },
      },
      getters: {
        menu: (state) => {
          return state.menu;
        },
      },
    });

    const wrapper = shallowMount(Sidebar, {
      store,
      localVue,
    });

    expect(wrapper.vm.isListener).toBeTruthy();
    expect(wrapper.vm.hasAccessAdminMenu).toBeFalsy();
    expect(wrapper.vm.isStudent).toBeFalsy();
  });

  it('computed property hasAccessAdminMenu is truthy, others are Falsy', () => {
    const store = new Vuex.Store({
      state: {
        menu: {
          admin: ['TEST'],
        },
      },
      getters: {
        menu: (state) => {
          return state.menu;
        },
      },
    });

    const wrapper = shallowMount(Sidebar, {
      store,
      localVue,
    });

    expect(wrapper.vm.isListener).toBeFalsy();
    expect(wrapper.vm.hasAccessAdminMenu).toBeTruthy();
    expect(wrapper.vm.isStudent).toBeFalsy();
  });
});
