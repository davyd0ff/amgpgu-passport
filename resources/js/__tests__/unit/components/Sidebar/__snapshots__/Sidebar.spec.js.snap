// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar.vue render 1`] = `
<aside>
  <ul
    class="sidenav sidenav-fixed grey-text text-darken-4"
    id="sidenav-left"
    style="transform: translateX(0px);"
  >
    <li>
      <user-info-stub />
    </li>
     
    <li
      class="no-padding"
    >
      <ul
        class="collapsible collapsible-accordion"
      >
        <!---->
         
        <!---->
         
        <!---->
      </ul>
    </li>
  </ul>
</aside>
`;
